ark/utils/spatial_lda_utils.py

```diff
@@ -1,63 +1,91 @@
-            to the index of each field of view which should be kept.
-
-    Returns:
-        None
     """
 
-    base_cols = [
-        "point",
-        "label",
-        "cell_size",
-        "centroid-0",
-        "centroid-1",
-        "pixelfreq_hclust_cap",
-        "name"
-    ]
-
-    # Check cell table
-    if not isinstance(cell_table, pd.DataFrame):
-        raise ValueError("cell_table must be a pd.DataFrame")
-
-    if not all([True for x in base_cols if x in cell_table.columns]):
-        raise ValueError("cell table must contain the following columns:{}".format(base_cols))
+    # Check cell table columns
+    verify_in_list(cell_table_columns=cell_table.columns.to_list(), required_columns=BASE_COLS)
 
     # Check markers/clusters
-    if all([markers is None, clusters is None]):
+    if markers is None and clusters is None:
         raise ValueError("markers and clusters cannot both be None")
     if markers is not None:
-        if isinstance(markers, list) and len(markers) == 0:
+        if len(markers) == 0:
             raise ValueError("list of marker names cannot be empty")
-        if not isinstance(markers, list) or not all([isinstance(x, str) for x in markers]):
-            raise ValueError("clusters must be a list of integers")
-        assert all([x in cell_table.columns for x in markers])
+        verify_in_list(markers=markers, cell_table_columns=cell_table.columns.to_list())
     if clusters is not None:
-        if isinstance(clusters, list) and len(clusters) == 0:
+        if len(clusters) == 0:
             raise ValueError("list of cluster ids cannot be empty")
-        if not isinstance(clusters, list) or not all([isinstance(x, int) for x in clusters]):
-            raise ValueError("markers must be a list of strings")
+        if not isinstance(clusters, list) or not all(
+                [isinstance(x, int) for x in clusters]):
+            raise TypeError("clusters must be a list of integers")
+
 
-    # Check fovs
-    if fovs != "all_fovs":
-        if not all([isinstance(x, int) for x in fovs]) or not isinstance(fovs, list) or len(fovs) == 0:
-            raise ValueError("fovs must be 'all_fovs' or a list of integers")
+def check_featurize_cell_table_args(cell_table, featurization, radius, cell_index):
+    """Checks the input arguments of the featurize_cell_table() function.
 
-    return None
+    Args:
+        cell_table (dict):
+            A dictionary whose elements are the correctly formatted dataframes for each field of
+            view.
+        featurization (str):
+            One of "cluster", "marker", "avg_marker", or "count".
+        radius (int):
+            Pixel radius corresponding to cellular neighborhood size.
+        cell_index (str):
+            Name of the column in each field of view pd.Dataframe indicating reference cells.
+    """
+    # Check valid data types
+    if not isinstance(cell_table, dict):
```

Comment:
I think we can assume that they passed the output of `format_cell_table` to this function, so don't need to check that it's a dict of dfs